scripts/tfsenc_encoding.py

```diff
@@ -0,0 +1,243 @@
+import csv
+import os
+import numpy as np
+import pandas as pd
+import torch
+from numba import jit, prange
+from scipy import stats
+from sklearn.preprocessing import StandardScaler
+from sklearn.decomposition import PCA
+from sklearn.linear_model import LinearRegression
+from sklearn.model_selection import GroupKFold, KFold
+from sklearn.pipeline import make_pipeline
+from himalaya.ridge import GroupRidgeCV, RidgeCV, ColumnTransformerNoStack
+from himalaya.kernel_ridge import (
+    MultipleKernelRidgeCV,
+    KernelRidgeCV,
+    ColumnKernelizer,
+    Kernelizer,
+)
+from himalaya.scoring import correlation_score, correlation_score_split
+
+
+@jit(nopython=True)
+def build_Y(brain_signal, onsets, lags, window_size):
+    """[summary]
+
+    Args:
+        onsets ([type]): [description]
+        brain_signal ([type]): [description]
+        lags ([type]): [description]
+        window_size ([type]): [description]
+
+    Returns:
+        [type]: [description]
+    """
+    half_window = round((window_size / 1000) * 512 / 2)  # convert to signal fs
```

Comment:
nit: for better code documentation, can we make 512 and 1000 named variables?

Reply:
sure if you like :)..they are constants anyway...

Reply:
and probably the only place where they are used...if you notice them being used elsewhere, makes sense to make them global constants ala prjct_id, sub_id, etc.